File: scrapers/humoruniv.py
import httpx

from scrapers.base import BaseScraper, Post


class HumorunivScraper(BaseScraper):
    community = "humoruniv"
    community_name = "웃긴대학"
    base_url = "http://web.humoruniv.com"
    encoding = "euc-kr"

    async def fetch_best_posts(self, client: httpx.AsyncClient) -> list[Post]:
        soup = await self._fetch_html(client, f"{self.base_url}/board/humor/board_best.html")
        posts = []

        for row in soup.select("table.list_table tr"):
            a_tag = row.select_one("a[href*='read.html']")
            if not a_tag:
                continue
            title = a_tag.get_text(strip=True)
            href = a_tag.get("href", "")
            if not href or not title:
                continue

            if href.startswith("http"):
                url = href
            elif href.startswith("/"):
                url = f"{self.base_url}{href}"
            else:
                # 상대 경로 (같은 디렉토리)
                url = f"{self.base_url}/board/humor/{href}"
            posts.append(self._make_post(title, url))

        # 대체: 좀 더 넓은 셀렉터
        if not posts:
            seen = set()
            for a_tag in soup.find_all("a", href=True):
                href = a_tag["href"]
                if "read.html" not in href:
                    continue
                title = a_tag.get_text(strip=True)
                if not title or len(title) < 2:
                    continue
                if href.startswith("http"):
                    url = href
                elif href.startswith("/"):
                    url = f"{self.base_url}{href}"
                else:
                    url = f"{self.base_url}/board/humor/{href}"
                if url not in seen:
                    seen.add(url)
                    posts.append(self._make_post(title, url))

        return posts

File: internal/scraper/humoruniv.go
package scraper

import (
	"fmt"
	"net/http"
	"strings"

	"github.com/PuerkitoBio/goquery"
)

// HumorunivScraper scrapes 웃긴대학 best (EUC-KR).
type HumorunivScraper struct {
	baseScraper
}

func NewHumorunivScraper() *HumorunivScraper {
	return &HumorunivScraper{
		baseScraper: baseScraper{
			community:     "humoruniv",
			communityName: "웃긴대학",
			baseURL:       "http://web.humoruniv.com",
			encoding:      "euc-kr",
		},
	}
}

func (s *HumorunivScraper) Name() string { return s.communityName }

func (s *HumorunivScraper) FetchBestPosts(client *http.Client) ([]Post, error) {
	doc, err := fetchDocument(client, fmt.Sprintf("%s/board/humor/board_best.html", s.baseURL), s.encoding, s.baseURL)
	if err != nil {
		return nil, err
	}

	var posts []Post
	seen := make(map[string]bool)

	// Each post title lives in <span id="title_chk_<table>-<number>">. The
	// enclosing <a> also wraps the comment-count span (.list_comment_num)
	// and a vote span, so reading the anchor text would fold
	// "[77] 답글추천 +481" into the title -- read the title span directly.
	doc.Find("span[id^='title_chk_']").Each(func(_ int, titleSpan *goquery.Selection) {
		title := strings.TrimSpace(titleSpan.Text())
		if title == "" {
			return
		}
		href, exists := titleSpan.Closest("a[href*='read.html']").Attr("href")
		if !exists || href == "" {
			return
		}

		var url string
		switch {
		case strings.HasPrefix(href, "http"):
			url = href
		case strings.HasPrefix(href, "/"):
			url = s.baseURL + href
		default:
			url = s.baseURL + "/board/humor/" + href
		}
		if seen[url] {
			return
		}
		seen[url] = true
		posts = append(posts, s.makePost(title, url, 0, 0, 0))
	})

	return s.filterPosts(posts), nil
}
